File: algorithms/floyd_warshall/java/floyd_warshall.java
/**
 * A Java program for Floyd Warshall All Pairs Shortest
 * Path algorithm.
 */
public class ShortestPath {
    private final static int VERTEX_NUMBER = 4;
    private final static int MAX_VALUE = 999999;

    public void floydWarshall(int graph[][]) {
        int dist[][] = new int[VERTEX_NUMBER][VERTEX_NUMBER];
        int i, j, k;

        for (i = 0; i < VERTEX_NUMBER; i++)
            for (j = 0; j < VERTEX_NUMBER; j++)
                dist[i][j] = graph[i][j];

        for (k = 0; k < VERTEX_NUMBER; k++) {
            for (i = 0; i < VERTEX_NUMBER; i++) {
                for (j = 0; j < VERTEX_NUMBER; j++) {
                    if (dist[i][k] + dist[k][j] < dist[i][j]) {
                        dist[i][j] = dist[i][k] + dist[k][j];
                    }
                }
            }
        }

        printSolution(dist);
    }

    private void printSolution(int dist[][]) {
        for (int i = 0; i < VERTEX_NUMBER; ++i) {
            for (int j = 0; j < VERTEX_NUMBER; ++j) {
                if (dist[i][j] == MAX_VALUE) {
                    System.out.print("INF ");
                } else {
                    System.out.print(dist[i][j] + "   ");
                }
            }
            System.out.println();
        }
    }

    /**
     * Test
     * @param args
     */
    public static void main (String[] args) {
        int graph[][] = {
                {0,         5,          MAX_VALUE,  10},
                {MAX_VALUE, 0,          3,          MAX_VALUE},
                {MAX_VALUE, MAX_VALUE,  0,          1},
                {MAX_VALUE, MAX_VALUE,  MAX_VALUE,  0}
        };

        ShortestPath a = new ShortestPath();

        a.floydWarshall(graph);
    }
}

File: algorithms/floyd_warshall/python/floyd-warshall.py
# Author: Parth Doshi
# Github ID: arsenal-2004

# This program is a Python 3.5 implementation of the Floyd-Warshall Shortest Path algorithm.

# Defining a value to be used as the distance between nodes without any direct edges
INF = 99999

# Assuming graph to be a square matrix
# In Python, that is represented as a list of lists.
def shortestPath(graph):
	
	# The number of vertices in the graph
	num_v = len(graph)

	# Making a copy of the graph to be used as the result variable
	dist = graph[:]

	for i in range(num_v):
		# Selecting a source vertex
		for j in range(num_v):
			# Selecting a destination vertex
			for k in range(num_v):
				dist[j][k] = min(dist[j][k], dist[j][i] + dist[i][k])

	return dist


def printSolution(dist): 
	for i in range(len(dist)):
		for j in range(len(dist)):
			print("{0:7d}".format(dist[i][j]), end="")
		print()

# Driver program to test the above program 
# Let us create the following weighted graph 
""" 
            10 
       (0)------->(3) 
        |         /|\
      5 |          | 
        |          | 1 
       \|/         | 
       (1)------->(2) 
            3           """

graph = [[0,5,INF,10], 
             [INF,0,3,INF], 
             [INF, INF, 0,   1], 
             [INF, INF, INF, 0] 
        ] 
# Print the solution 
dist = shortestPath(graph)	
printSolution(dist)	
